src/extractor/extraction_containers.cpp

```diff
@@ -453,11 +453,13 @@ void ExtractionContainers::PrepareEdges(lua_State *segment_state)
                 all_edges_list[i].result.weight < min_forward_weight)
             {
                 min_forward_idx = i;
+                min_forward_weight = all_edges_list[i].result.weight;
```

Comment:
Yes, always the last edge was used. Because the order of edges was (source, target),  edge weights were "randomized" by scheduler.